src/esn.calendar.libs/app/services/partstat-jwt.service.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const jwtDecode = require('jwt-decode');
+
+angular.module('esn.calendar.libs')
+  .service('calPartstatJwtService', calPartstatJwtService);
+
+function calPartstatJwtService($q, $http, $log, httpConfigurer, calEventService, calendarHomeService, calOpenEventForm, notificationFactory, calPartstatUpdateNotificationService) {
+  return {
+    changeParticipationUsingJwtAndDisplayEvent,
+    changeParticipationUsingJwt,
+    getChangeParticipationUrl
+  };
+
+  /**
+   * Change a user's participation status for an event using a jwt and display the event details afterwards
+   * @param {String} jwt    The jwt that contains the information to change the user's participation status for an event
+   * @returns {Promise}     A promise that resolves when the process to change the participation status using a jwt finishes
+   */
+  function changeParticipationUsingJwtAndDisplayEvent(jwt) {
+    if (!jwt) {
+      const error = new Error('There is no jwt to change the participation status');
+
+      $log.error(error);
+      notificationFactory.weakError('', 'Event participation modification failed');
+
+      return $q.reject(error);
+    }
+
+    const { uid: eventUid, action } = jwtDecode.default(jwt);
```

Comment:
Is it safer if we check if `jwtDecode.default(jwt)` is defined?

Reply:
It will always be defined unless we import the wrong module :trollface:.